01-JS-DOM-APIs/Ex03/index.html

```diff
@@ -1,33 +1,45 @@
 <!DOCTYPE html>
 <html>
-   <head>
+  <head>
 		<meta charset="UTF-8">
 		<meta http-equiv="x-ua-compatible" content="ie=edge">
    	<title> JS 03 </title>
    	<meta name="description" content="Let's begin with JavaScript">
    	<meta name="viewport" content="width=device-width, initial-scale=1">
-      <link rel="stylesheet" href="css/styles.css">
-      <link href="https://fonts.googleapis.com/css?family=Bubbler+One" rel="stylesheet">
-   </head>
-   <body>
-   	<header>
-   		<h1>This it my third JS excercise!</h1>
-   	</header>
-   	<section id="jokeTextbox"></section> <br>
-      <button onclick="getJoke('http://api.icndb.com/jokes/random')"> I have a joke for you </button>
-      <script>
-         function getJoke(url) {
-            var joke;
-            var jokeRequest = new XMLHttpRequest();
-            jokeRequest.onreadystatechange = function() {
-            if (this.readyState == 4 && this.status == 200) {
-               joke=JSON.parse(this.responseText);
-               document.getElementById("jokeTextbox").innerHTML = joke.value.joke;
+    <link rel="stylesheet" href="css/styles.css">
+    <link href="https://fonts.googleapis.com/css?family=Bubbler+One" rel="stylesheet">
+  </head>
+  <body>
+    <header>
+     	<h1>This it my third JS excercise!</h1>
+    </header>
+    <section id="jokeTextbox"></section> <br>
+    <button onclick="getJoke (jconfig)"> I have a joke for you </button>
+    <script>
+      var jconfig = {
+        fmethod:'GET',
+        url:'http://api.icndb.com/jokes/random',
+        sync:true
+      }
+      function getJoke (config) {
+        let promise = new Promise (function(resolve, reject) {
+          let jokeRequest = new XMLHttpRequest();
+          jokeRequest.open(config.fmethod, config.url, config.sync);
```

Comment:
If I see `getJoke` as a black box, the input and the output are the expected. Let's analyze the requirement together:

First, create a function that returns a joke, 

```
let getJoke = function () {
//something to get the joke
return joke;
}

```
Then, create a reusable function that perfroms AJAX calls. That function will receive a config object and return a promise.

```
let performAjaxCall = function (config) {
     return new Promise(this.promiseHandler); // the promise should be 
}

let promiseHandler = function (resolve, reject) {
     //We should manipulate the promise here, determine when we should resolve or reject it
};
```

Give a second try and let me know about the results.